data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/PostBlockV2.java

```diff
@@ -0,0 +1,148 @@
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult.FailureReason;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionCache;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitions;
+
+public class PostBlockV2 extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v2/beacon/blocks";
+
+  private final ValidatorDataProvider validatorDataProvider;
+  private final SyncDataProvider syncDataProvider;
+
+  public PostBlockV2(
+      final DataProvider dataProvider,
+      final Spec spec,
+      final SchemaDefinitionCache schemaDefinitionCache) {
+    super(createMetadata(spec, schemaDefinitionCache));
+    this.validatorDataProvider = dataProvider.getValidatorDataProvider();
+    this.syncDataProvider = dataProvider.getSyncDataProvider();
+  }
+
+  PostBlockV2(
+      final ValidatorDataProvider validatorDataProvider,
+      final SyncDataProvider syncDataProvider,
+      final Spec spec,
+      final SchemaDefinitionCache schemaDefinitionCache) {
+    super(createMetadata(spec, schemaDefinitionCache));
+    this.validatorDataProvider = validatorDataProvider;
+    this.syncDataProvider = syncDataProvider;
+  }
+
+  private static EndpointMetadata createMetadata(
+      final Spec spec, final SchemaDefinitionCache schemaDefinitionCache) {
+    return EndpointMetadata.post(ROUTE)
+        .operationId("publishBlockV2")
+        .summary("Publish a signed block")
+        .description(
+            """
+            Instructs the beacon node to broadcast a newly signed beacon block to the beacon network, \
+            to be included in the beacon chain. A success response (20x) indicates that the block \
+            passed gossip validation and was successfully broadcast onto the network. \
+            The beacon node is also expected to integrate the block into the state, but may broadcast it \
+            before doing so, so as to aid timely delivery of the block. Should the block fail full \
+            validation, a separate success response code (202) is used to indicate that the block was \
+            successfully broadcast but failed integration. After Deneb, this additionally instructs \
+            the beacon node to broadcast all given signed blobs. The broadcast behaviour may be adjusted via the \
+            `broadcast_validation` query parameter.""")
+        .tags(TAG_BEACON, TAG_VALIDATOR_REQUIRED)
+        .queryParam(PARAMETER_BROADCAST_VALIDATION)
+        .requestBodyType(
+            getSchemaDefinitionForAllSupportedMilestones(
+                schemaDefinitionCache,
+                "SignedBlock",
+                SchemaDefinitions::getSignedBlockContainerSchema,
+                (blockContainer, milestone) ->
+                    schemaDefinitionCache
+                        .milestoneAtSlot(blockContainer.getSlot())
+                        .equals(milestone)),
+            context ->
+                slotBasedSelector(
```

Comment:
In spec we have `Eth-Consensus-Version`. As I understand it's not really needed, yeah?

Reply:
ahh good catch... looking into it

Reply:
mm let's implement it in a separate PR

Reply:
to respond to your question, yes it isn't needed